index.js

```diff
@@ -408,9 +414,15 @@ class GithubScm extends Scm {
                   + 'then $SD_GIT_WRAPPER '
                   + `"git clone --recursive --quiet --progress --branch ${branch} `
                   + '$SCM_URL $SD_SOURCE_DIR"; '
-                  + 'else $SD_GIT_WRAPPER '
-                  + '"git clone --depth=50 --no-single-branch --recursive --quiet --progress '
-                  + `--branch ${branch} $SCM_URL $SD_SOURCE_DIR"; fi`);
+                  + 'else if [ -z $GIT_SHALLOW_CLONE_DEPTH ]; '
+                  + 'then export GIT_SHALLOW_CLONE_DEPTH=50; fi; '
```

Comment:
can this logic be defined outside ? so that we don't end up duplicating these lines ?